src/components/DataDictionary.tsx

```diff
@@ -126,6 +155,121 @@ export const DataDictionary: React.FC<{}> = () => {
   const [sidebarOpen, setSidebarOpen] = React.useState(true)
   const [search, setSearch] = React.useState('')
   const { currentExplore, loadingExplore } = useCurrentExplore()
+  const extensionContext = useContext<ExtensionContextData>(ExtensionContext)
+  const { extensionSDK, initializeError } = extensionContext
+  const [canPersistContextData, setCanPersistContextData] = useState<boolean>(
+    false
+  )
+  const { coreSDK } = useContext(ExtensionContext)
+  const [contextCopy, setContextCopy] = React.useState("{}")
+  const [authors, setAuthors] = React.useState<IUser[]>([])
+  const [comments, setComments] = React.useState("{}")
+  const [me, setMe] = React.useState<IUser>()
+
+
+  const addComment = async (newCommentStr: string, field: string): Promise<boolean> => {
+    let revivedComments = JSON.parse(contextCopy)
+    let newComment = JSON.parse(newCommentStr)
+    
+    if (revivedComments[currentExplore.name][field]) {
+      let revivedFields = revivedComments[currentExplore.name][field]
+      revivedFields.push(newComment)
+      revivedComments[currentExplore.name][field] = revivedFields
+    } else {
+      revivedComments[currentExplore.name][field] = [newComment]
+    }
+
+    if (canPersistContextData) {
+      try {
+        console.log(revivedComments)
+        setComments(JSON.stringify(revivedComments))
+        setContextCopy(JSON.stringify(revivedComments))
```

Comment:
They are the same thing. contextCopy could be removed in favor of comments, but it felt safer to do the write back to extension context using a copy that hasn't been touched since initial read.